File: LeetCode/Bit/Golang/Leetcode868-二进制间距.go
package sbgo

func binaryGap(n int) int {
	res, cur, pre := 0, 0, -1
	for n != 0 {
		if (n & 1) == 1 {
			if pre != -1 {
				if cur-pre > res {
					res = cur - pre
				}
			}

			pre = cur
		}
		n >>= 1
		cur++
	}
	return res
}

File: LeetCode/Bit/JS/Leetcode868-二进制间距.js
var binaryGap = function (n) {
    let [res, cur, pre] = [0, 0, -1];
    while (n) {
        if (n & 1) {
            if (pre != -1) {
                res = Math.max(res, cur - pre);
            }
            pre = cur;
        }

        n >>= 1;
        cur++;
    }
    return res;
};